### PR TITLE
add params id fork

### PR DESCRIPTION
**Pull request by dylangl**
add params id fork

### CODE REVIEW

**Comment by dylangl** [comment]
I want to use namespace in fork project api ,so I added params in it。

File: lib/Gitlab/Api/Projects.php [hunk 1]
```diff
@@ -498,9 +498,9 @@ public function removeLabel($project_id, $name)
      * @param int $project_id
      * @return mixed
      */
-    public function fork($project_id)
+    public function fork($project_id, array $params)
```
Comment (m1guelpf) [inline lib/Gitlab/Api/Projects.php]: The options parameter should be optional

File: lib/Gitlab/Api/Projects.php [hunk 2]
```diff
@@ -496,11 +496,12 @@ public function removeLabel($project_id, $name)
...
     /**
      * @param int $project_id
+     * @param array $params
      * @return mixed
      */
-    public function fork($project_id)
+    public function fork($project_id, array $params = array())
     {
-        return $this->post('projects/'.$this->encodePath($project_id).'/fork');
+        return $this->post('projects/'.$this->encodePath($project_id).'/fork', $params);
```
Comment (fbourigault) [inline lib/Gitlab/Api/Projects.php]: Could you add `$params` validation using `OptionResolver`?

Comment (dylangl) [reply lib/Gitlab/Api/Projects.php]: I don't know how to use;

**Comment by dylangl** [comment]
So,can you merge my pull requests?

File: lib/Gitlab/Api/Projects.php (hunk 2 above)
Comment (fbourigault) [reply lib/Gitlab/Api/Projects.php]: Here is an example of how to use `OptionResolver` for a `post` request: https://github.com/m4tthumphrey/php-gitlab-api/blob/032b7cca7b376686522d00bcb404bc9167a6e0c4/lib/Gitlab/Api/RepositoryFiles.php#L35-L68.

File: lib/Gitlab/Api/Projects.php [hunk 3]
```diff
@@ -496,11 +496,15 @@ public function removeLabel($project_id, $name)
...
+     *
+     *     @var string $namespace      The ID or path of the namespace that the project will be forked to
+     * )
      * @return mixed
      */
-    public function fork($project_id)
+    public function fork($project_id, array $params = array())
     {
-        return $this->post('projects/'.$this->encodePath($project_id).'/fork');
+        return $this->post($this->getProjectPath($project_id, 'fork'), $params);
```
Comment (fbourigault) [inline lib/Gitlab/Api/Projects.php]: Please, use the `OptionResolver` to add input data validation.

Comment (dylangl) [reply lib/Gitlab/Api/Projects.php]: OK

**Comment by dylangl** [comment]
I have finished it.

File: lib/Gitlab/Api/Projects.php [hunk 4]
```diff
@@ -496,11 +496,20 @@ public function removeLabel($project_id, $name)
...
     /**
      * @param int $project_id
+     * @param array $params (
+     *
+     *     @var string $namespace      The ID or path of the namespace that the project will be forked to
+     * )
      * @return mixed
      */
-    public function fork($project_id)
+    public function fork($project_id, array $params = array())
```
Comment (fbourigault) [inline lib/Gitlab/Api/Projects.php]: Could you use `$parameters` for consistency with other methods in this library?

File: lib/Gitlab/Api/Projects.php [hunk 5]
```diff
@@ -496,11 +496,20 @@ public function removeLabel($project_id, $name)
...
     {
-        return $this->post('projects/'.$this->encodePath($project_id).'/fork');
+        $resolver = $this->createOptionsResolver();
```
Comment (fbourigault) [inline lib/Gitlab/Api/Projects.php]: You can use `$this->createOptionsResolver()` here as this methods returns an `OptionsResolver` preconfigured for pagination. Here a `new OptionsResolver()` will do the job.

Comment (fbourigault) [reply lib/Gitlab/Api/Projects.php]: It looks like you missed this one ;)

**Comment by dylangl** [comment]
sorry,I didn't find it。	
Can you tell me where it is?

**Comment by fbourigault** [comment]
> Can you tell me where it is?

https://github.com/dylangl/php-gitlab-api/blob/4693d29f03497917c154930d1f7291f23a9b5259/lib/Gitlab/Api/Projects.php#L507

**Comment by dylangl** [comment]
I am looking here to write。
https://github.com/dylangl/php-gitlab-api/blob/4693d29f03497917c154930d1f7291f23a9b5259/lib/Gitlab/Api/Projects.php#L306

**Comment by fbourigault** [comment]
The `hooks` method is about fetching a collection of existing hooks, so pagination is enabled and `$this->createOptionsResolver` can be used here as it come with `page` and `per_page` defaults.
The `fork` is an action to fork the project into an other namespace, thus you can't use the `$this->createOptionsResolver()` because it would add the `page` and `per_page` options which are not valid in this case. You have to do it like in https://github.com/dylangl/php-gitlab-api/blob/4693d29f03497917c154930d1f7291f23a9b5259/lib/Gitlab/Api/RepositoryFiles.php#L53.

**Comment by dylangl** [comment]
use `$resolver = new OptionsResolver();`

**Comment by fbourigault** [comment]
Thank you!

**Pull request merged**